Sources/Moya/MoyaProvider.swift

```diff
@@ -150,11 +150,12 @@ public func convertResponseToResult(_ response: HTTPURLResponse?, request: URLRe
         case let (.some(response), data, .none):
             let response = Moya.Response(statusCode: response.statusCode, data: data ?? Data(), request: request, response: response)
             return .success(response)
-        case let (_, _, .some(error)):
-            let error = MoyaError.underlying(error)
+        case let (.some(response), _, .some(error)):
+            let response = Moya.Response(statusCode: response.statusCode, data: data ?? Data(), request: request, response: response)
```

Comment:
Do we still need to handle the `(_, _, .some(error))` case here in case we get an `AFError` that doesn't have a `Response`?